src/lib/containers/QueryWrapper.tsx

```diff
@@ -375,6 +387,7 @@ export default class QueryWrapper extends React.Component<
             }
           })
         }
+        const isFacetAvailable = isFacetAvailableAndSupported(this.props.initQueryRequest.query.sql, data.facets)
```

Comment:
On initialization, respect the default showFacetFilter and showFacetVisualization value.  But if configured to show, then only show if facets are actually available.